### PR TITLE
Allow importmap overrides in shim mode with option

### PR DESCRIPTION
**Pull request by lewisl9029**
As discussed in #254, this PR adds a `override` boolean option to allow overriding of existing import map entries in shim mode, defaulting to false to preserve current behavior. Overall increase in footprint is 7 bytes to 9167 according to the `npm run footprint` command.

A few hoops I had to jump through:

- Depending on ./config.js in ./common.js introduced a circular dependency, which caused some weird issues in tests. So I ended up refactoring the noop function into its own module to untangle it. I also tried inlining the noop function everywhere, but it ended up making the bundle larger than using a separate module.

- For testing, I ended up adding a separate module for this option because I couldn't figure out a way to make the test coexist in the same module without conflicting with the existing importmap overrides tests, which require this option to be off. Open to alternatives!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -432,6 +433,7 @@ window.esmsInitOptions = {
   fetch: (url, options) => fetch(url, options), // default is native
   revokeBlobURLs: true, // default false
   enforceIntegrity: true, // default false
+  override: true, // default false
```
Comment (guybedford) [inline README.md:436]: Let's give it a more descriptive name, perhaps something like `mapOverrides: true`?

File: src/common.js [hunk 2]
```diff
@@ -113,7 +113,7 @@ export function resolveUrl (relUrl, parentUrl) {
 function resolveAndComposePackages (packages, outPackages, baseUrl, parentMap) {
   for (let p in packages) {
     const resolvedLhs = resolveIfNotPlainOrUrl(p, baseUrl) || p;
-    if (outPackages[resolvedLhs] && (outPackages[resolvedLhs] !== packages[resolvedLhs])) {
+    if ((!shimMode || !override) && outPackages[resolvedLhs] && (outPackages[resolvedLhs] !== packages[resolvedLhs])) {
```
Comment (guybedford) [inline src/common.js:116]: Rather than moving the "noop" to a separate file, what about creating a separate `resolve.js` module?

**Review by guybedford — APPROVED** [review]
Looks great, thanks for your work on this!

File: README.md (hunk 1 above)
Comment (lewisl9029) [reply README.md:436]: Yep, sounds good, went a bit overboard with the bytes saving here. 😅

File: src/common.js (hunk 2 above)
Comment (lewisl9029) [reply src/common.js:116]: Good call, let me give it a shot!

Comment (lewisl9029) [reply src/common.js]: Addressed here: https://github.com/guybedford/es-module-shims/pull/257/commits/8140a862d64fa661533b6f4ea2367044bef8724d

File: README.md (hunk 1 above)
Comment (lewisl9029) [reply README.md:436]: Addressed here: https://github.com/guybedford/es-module-shims/pull/257/commits/39432375f5d59f61271f1d5bf8b0360322c4a54d

**Comment by lewisl9029** [comment]
With the latest changes we're only 2 bytes higher at 9162 now! I guess refactoring out the resolve functions must have saved enough bytes to make up for everything else, somehow.

**Comment by guybedford** [comment]
Thanks for the quick turnaround on this one - footprint is great!

Will release soon, just hoping to get https://github.com/guybedford/es-module-shims/issues/255 fix in as well.

**Pull request merged**